--- ./VirtualBox-kmod-6.0.2/vboxsf/vfsmod.c.orig	2019-01-27 03:29:18.138159346 +0000
+++ ./VirtualBox-kmod-6.0.2/vboxsf/vfsmod.c	2019-01-27 03:54:08.850764196 +0000
@@ -43,6 +43,9 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0)
 # include <linux/mount.h>
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
+# include <uapi/linux/mount.h>
+#endif
 #include <linux/seq_file.h>
 
 MODULE_DESCRIPTION(VBOX_PRODUCT " VFS Module for Host File System Access");
