diff -rup VirtualBox-kmod-6.1.26.orig/vboxdrv/include/iprt/stdarg.h VirtualBox-kmod-6.1.26/vboxdrv/include/iprt/stdarg.h
--- VirtualBox-kmod-6.1.26.orig/vboxdrv/include/iprt/stdarg.h	2021-07-28 17:13:57.000000000 +0100
+++ VirtualBox-kmod-6.1.26/vboxdrv/include/iprt/stdarg.h	2021-09-13 13:04:24.536235200 +0100
@@ -44,7 +44,12 @@
 #   define __builtin_stdarg_start __builtin_va_start
 #  endif
 # else
-#  include <stdarg.h>
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+#   include <stdarg.h>
+#  else
+#   include <linux/stdarg.h>
+#  endif
 # endif
 #endif
 
diff -rup VirtualBox-kmod-6.1.26.orig/vboxguest/include/iprt/stdarg.h VirtualBox-kmod-6.1.26/vboxguest/include/iprt/stdarg.h
--- VirtualBox-kmod-6.1.26.orig/vboxguest/include/iprt/stdarg.h	2021-07-28 17:13:57.000000000 +0100
+++ VirtualBox-kmod-6.1.26/vboxguest/include/iprt/stdarg.h	2021-09-13 13:04:10.801129474 +0100
@@ -44,7 +44,12 @@
 #   define __builtin_stdarg_start __builtin_va_start
 #  endif
 # else
-#  include <stdarg.h>
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+#   include <stdarg.h>
+#  else
+#   include <linux/stdarg.h>
+#  endif
 # endif
 #endif
 
diff -rup VirtualBox-kmod-6.1.26.orig/vboxnetadp/include/iprt/stdarg.h VirtualBox-kmod-6.1.26/vboxnetadp/include/iprt/stdarg.h
--- VirtualBox-kmod-6.1.26.orig/vboxnetadp/include/iprt/stdarg.h	2021-07-28 17:13:57.000000000 +0100
+++ VirtualBox-kmod-6.1.26/vboxnetadp/include/iprt/stdarg.h	2021-09-13 13:04:05.972092303 +0100
@@ -44,7 +44,12 @@
 #   define __builtin_stdarg_start __builtin_va_start
 #  endif
 # else
-#  include <stdarg.h>
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+#   include <stdarg.h>
+#  else
+#   include <linux/stdarg.h>
+#  endif
 # endif
 #endif
 
diff -rup VirtualBox-kmod-6.1.26.orig/vboxnetflt/include/iprt/stdarg.h VirtualBox-kmod-6.1.26/vboxnetflt/include/iprt/stdarg.h
--- VirtualBox-kmod-6.1.26.orig/vboxnetflt/include/iprt/stdarg.h	2021-07-28 17:13:57.000000000 +0100
+++ VirtualBox-kmod-6.1.26/vboxnetflt/include/iprt/stdarg.h	2021-09-13 13:04:17.661182279 +0100
@@ -44,7 +44,12 @@
 #   define __builtin_stdarg_start __builtin_va_start
 #  endif
 # else
-#  include <stdarg.h>
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+#   include <stdarg.h>
+#  else
+#   include <linux/stdarg.h>
+#  endif
 # endif
 #endif
 
diff -rup VirtualBox-kmod-6.1.26.orig/vboxsf/include/iprt/stdarg.h VirtualBox-kmod-6.1.26/vboxsf/include/iprt/stdarg.h
--- VirtualBox-kmod-6.1.26.orig/vboxsf/include/iprt/stdarg.h	2021-07-28 17:13:57.000000000 +0100
+++ VirtualBox-kmod-6.1.26/vboxsf/include/iprt/stdarg.h	2021-09-13 13:04:30.755283071 +0100
@@ -44,7 +44,12 @@
 #   define __builtin_stdarg_start __builtin_va_start
 #  endif
 # else
-#  include <stdarg.h>
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+#   include <stdarg.h>
+#  else
+#   include <linux/stdarg.h>
+#  endif
 # endif
 #endif
 
