diff -rup VirtualBox-kmod-6.0.8/vboxdrv/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxdrv/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxdrv/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxdrv/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
diff -rup VirtualBox-kmod-6.0.8/vboxguest/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxguest/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxguest/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxguest/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
diff -rup VirtualBox-kmod-6.0.8/vboxnetadp/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxnetadp/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxnetadp/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxnetadp/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
diff -rup VirtualBox-kmod-6.0.8/vboxnetflt/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxnetflt/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxnetflt/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxnetflt/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
diff -rup VirtualBox-kmod-6.0.8/vboxpci/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxpci/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxpci/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxpci/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
diff -rup VirtualBox-kmod-6.0.8/vboxsf/Makefile-header.gmk VirtualBox-kmod-6.0.8.new/vboxsf/Makefile-header.gmk
--- VirtualBox-kmod-6.0.8/vboxsf/Makefile-header.gmk	2019-05-13 14:33:15.000000000 +0100
+++ VirtualBox-kmod-6.0.8.new/vboxsf/Makefile-header.gmk	2019-05-31 03:03:55.997181820 +0100
@@ -127,7 +127,7 @@ ifeq ($(KERNELRELEASE),)
    VBOX_KERN_GROKS_EXTMOD :=
   endif
   VBOX_KERN_GROKS_SUBDIRS  :=
-  ifeq ($(shell if grep '"[432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
+  ifeq ($(shell if grep '"[5432]\.' $(KERN_DIR)/include/linux/version.h > /dev/null 2>&1; then echo yes; fi),yes)
    VBOX_KERN_GROKS_SUBDIRS := yes
   endif
  endif
