diff -rup VirtualBox-kmod-7.2.12.orig/vboxnetadp/VBoxNetAdp.c VirtualBox-kmod-7.2.12/vboxnetadp/VBoxNetAdp.c
--- VirtualBox-kmod-7.2.12.orig/vboxnetadp/VBoxNetAdp.c	2026-06-26 15:23:24.000000000 +0100
+++ VirtualBox-kmod-7.2.12/vboxnetadp/VBoxNetAdp.c	2026-07-09 01:10:51.009931504 +0100
@@ -129,8 +129,7 @@ int vboxNetAdpCreate(PVBOXNETADP *ppNew,
             {
                 Log(("vboxNetAdpCreate: using name: %s\n", pcszName));
                 pThis->iUnit = vboxNetAdpGetUnitByName(pcszName);
-                strncpy(pThis->szName, pcszName, sizeof(pThis->szName) - 1);
-                pThis->szName[sizeof(pThis->szName) - 1] = '\0';
+                strscpy(pThis->szName, pcszName, sizeof(pThis->szName));
             }
             else
             {
diff -rup VirtualBox-kmod-7.2.12.orig/vboxnetadp/linux/VBoxNetAdp-linux.c VirtualBox-kmod-7.2.12/vboxnetadp/linux/VBoxNetAdp-linux.c
--- VirtualBox-kmod-7.2.12.orig/vboxnetadp/linux/VBoxNetAdp-linux.c	2026-06-26 15:23:25.000000000 +0100
+++ VirtualBox-kmod-7.2.12/vboxnetadp/linux/VBoxNetAdp-linux.c	2026-07-09 01:09:58.080214893 +0100
@@ -342,8 +342,7 @@ int vboxNetAdpOsCreate(PVBOXNETADP pThis
             err = register_netdev(pNetDev);
             if (!err)
             {
-                strncpy(pThis->szName, pNetDev->name, sizeof(pThis->szName));
-                pThis->szName[sizeof(pThis->szName) - 1] = '\0';
+                strscpy(pThis->szName, pNetDev->name, sizeof(pThis->szName));
                 pThis->u.s.pNetDev = pNetDev;
                 Log2(("vboxNetAdpOsCreate: pThis=%p pThis->szName = %p\n", pThis, pThis->szName));
                 return VINF_SUCCESS;
@@ -465,8 +464,7 @@ static long VBoxNetAdpLinuxIOCtlUnlocked
             }
 
             Assert(strlen(pAdp->szName) < sizeof(Req.szName));
-            strncpy(Req.szName, pAdp->szName, sizeof(Req.szName) - 1);
-            Req.szName[sizeof(Req.szName) - 1] = '\0';
+            strscpy(Req.szName, pAdp->szName, sizeof(Req.szName));
 
             if (RT_UNLIKELY(copy_to_user((void *)ulArg, &Req, sizeof(Req))))
             {
diff -rup VirtualBox-kmod-7.2.12.orig/vboxsf/vfsmod.c VirtualBox-kmod-7.2.12/vboxsf/vfsmod.c
--- VirtualBox-kmod-7.2.12.orig/vboxsf/vfsmod.c	2026-06-26 15:08:13.000000000 +0100
+++ VirtualBox-kmod-7.2.12/vboxsf/vfsmod.c	2026-07-09 01:12:50.983661862 +0100
@@ -548,8 +548,7 @@ static void vbsf_init_mount_info(struct
     mount_info->length = sizeof(struct vbsf_mount_info_new);
     if (sf_name) {
 # if RTLNX_VER_MAX(2,5,69)
-        strncpy(mount_info->name, sf_name, sizeof(mount_info->name));
-        mount_info->name[sizeof(mount_info->name)-1] = 0;
+        strscpy(mount_info->name, sf_name, sizeof(mount_info->name));
 # else
         strlcpy(mount_info->name, sf_name, sizeof(mount_info->name));
 # endif
@@ -765,8 +764,7 @@ static int vbsf_parse_mount_options(char
         if (!strcmp(option, "iocharset") || !strcmp(option, "nls")) {
             if (!value || !*value)
                 return -EINVAL;
-            strncpy(mount_info->nls_name, value, sizeof(mount_info->nls_name));
-            mount_info->nls_name[sizeof(mount_info->nls_name)-1] = 0;
+            strscpy(mount_info->nls_name, value, sizeof(mount_info->nls_name));
         } else if (!strcmp(option, "uid")) {
             mount_info->uid = simple_strtoul(value, &value, 0);
             if (*value)
@@ -832,13 +830,11 @@ static int vbsf_parse_mount_options(char
         } else if (!strcmp(option, "tag")) {
             if (!value || !*value)
                 return -EINVAL;
-            strncpy(mount_info->szTag, value, sizeof(mount_info->szTag));
-            mount_info->szTag[sizeof(mount_info->szTag)-1] = 0;
+            strscpy(mount_info->szTag, value, sizeof(mount_info->szTag));
         } else if (!strcmp(option, "sf_name")) {
             if (!value || !*value)
                 return -EINVAL;
-            strncpy(mount_info->name, value, sizeof(mount_info->name));
-            mount_info->name[sizeof(mount_info->name)-1] = 0;
+            strscpy(mount_info->name, value, sizeof(mount_info->name));
         } else {
             printk(KERN_ERR "unrecognised mount option \"%s\"", option);
             return -EINVAL;
