diff -rup ./VirtualBox-kmod-5.1.22.orig/vboxdrv/r0drv/linux/waitqueue-r0drv-linux.h ./VirtualBox-kmod-5.1.22/vboxdrv/r0drv/linux/waitqueue-r0drv-linux.h
--- ./VirtualBox-kmod-5.1.22.orig/vboxdrv/r0drv/linux/waitqueue-r0drv-linux.h	2017-04-28 16:04:49.000000000 +0100
+++ ./VirtualBox-kmod-5.1.22/vboxdrv/r0drv/linux/waitqueue-r0drv-linux.h	2017-07-12 15:49:45.479872750 +0100
@@ -39,6 +39,9 @@
  *  schedule_hrtimeout_range. */
 #define RTR0SEMLNXWAIT_RESOLUTION   50000
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 13, 0)
+#define wait_queue_t wait_queue_entry_t
+#endif
 
 /**
  * Kernel mode Linux wait state structure.
diff -rup ./VirtualBox-kmod-5.1.22.orig/vboxguest/r0drv/linux/waitqueue-r0drv-linux.h ./VirtualBox-kmod-5.1.22/vboxguest/r0drv/linux/waitqueue-r0drv-linux.h
--- ./VirtualBox-kmod-5.1.22.orig/vboxguest/r0drv/linux/waitqueue-r0drv-linux.h	2017-04-28 16:04:49.000000000 +0100
+++ ./VirtualBox-kmod-5.1.22/vboxguest/r0drv/linux/waitqueue-r0drv-linux.h	2017-07-12 15:49:45.479872750 +0100
@@ -39,6 +39,9 @@
  *  schedule_hrtimeout_range. */
 #define RTR0SEMLNXWAIT_RESOLUTION   50000
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 13, 0)
+#define wait_queue_t wait_queue_entry_t
+#endif
 
 /**
  * Kernel mode Linux wait state structure.
